fvm/src/kernel/default.rs

```diff
@@ -815,6 +816,15 @@ where
         actor_id: ActorID,
         delegated_address: Option<Address>,
     ) -> Result<()> {
+        if self.caller != INIT_ACTOR_ID {
```

Comment:
I think you're looking for `self.actor_id` here, since `caller` would be the ActorID of the actor that immediately called the actor that's currently executing (the _caller_ in this context).